rpcserver.go

```diff
@@ -1040,6 +1064,133 @@ func (r *rpcServer) AddrReceives(ctx context.Context,
+	}
+
+	raw := in.GetRaw()
+	if len(raw.Inputs) > 0 {
+		return nil, fmt.Errorf("template inputs not yet supported")
+	}
+	if len(raw.Recipients) > 1 {
+		return nil, fmt.Errorf("only one recipient supported")
+	}
+
+	var (
+		taroParams = address.ParamsForChain(r.cfg.ChainParams.Name)
+		addr       *address.Taro
+		err        error
+	)
+	for a := range raw.Recipients {
+		addr, err = address.DecodeAddress(a, &taroParams)
+		if err != nil {
+			return nil, fmt.Errorf("unable to decode addr: %w", err)
+		}
+	}
+
+	if addr == nil {
+		return nil, fmt.Errorf("no recipients specified")
+	}
+
+	vPacket, _, err := r.cfg.AssetWallet.FundAddressSend(ctx, *addr)
+	if err != nil {
+		return nil, fmt.Errorf("error funding address send: %w", err)
+	}
+
+	var b bytes.Buffer
+	if err := vPacket.Serialize(&b); err != nil {
+		return nil, fmt.Errorf("error serializing packet: %w", err)
+	}
+
+	return &wrpc.FundVirtualPsbtResponse{
+		FundedPsbt:        b.Bytes(),
+		ChangeOutputIndex: 0,
+	}, nil
+}
+
+// SignVirtualPsbt signs the inputs of a virtual transaction and prepares the
+// commitments of the inputs and outputs.
+func (r *rpcServer) SignVirtualPsbt(_ context.Context,
+	in *wrpc.SignVirtualPsbtRequest) (*wrpc.SignVirtualPsbtResponse,
+	error) {
+
+	vPacket, err := taropsbt.NewFromRawBytes(
+		bytes.NewReader(in.FundedPsbt), false,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding packet: %w", err)
+	}
+
+	err = r.cfg.AssetWallet.SignVirtualPacket(vPacket)
+	if err != nil {
+		return nil, fmt.Errorf("error signing packet: %w", err)
+	}
+
+	var b bytes.Buffer
+	if err := vPacket.Serialize(&b); err != nil {
+		return nil, fmt.Errorf("error serializing packet: %w", err)
+	}
+
+	return &wrpc.SignVirtualPsbtResponse{
+		SignedPsbt:   b.Bytes(),
+		SignedInputs: []uint32{0},
+	}, nil
+}
+
+// AnchorVirtualPsbts merges and then commits multiple virtual transactions in
+// a single BTC level anchor transaction.
+//
+// TODO(guggero): Actually implement accepting and merging multiple
+// transactions.
+func (r *rpcServer) AnchorVirtualPsbts(ctx context.Context,
+	in *wrpc.AnchorVirtualPsbtsRequest) (*tarorpc.SendAssetResponse,
+	error) {
+
+	if len(in.VirtualPsbts) > 1 {
+		return nil, fmt.Errorf("only one virtual PSBT supported")
+	}
+
+	vPacket, err := taropsbt.NewFromRawBytes(
+		bytes.NewReader(in.VirtualPsbts[0]), false,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding packet: %w", err)
+	}
+
+	inputAsset := vPacket.Input.Asset()
+	prevID := vPacket.Input.PrevID
+	inputCommitment, err := r.cfg.AssetStore.FetchCommitment(
+		ctx, inputAsset.ID(), prevID.OutPoint, inputAsset.Amount,
+		inputAsset.GroupKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching input commitment: %w",
+			err)
+	}
+
+	rpcsLog.Debugf("Selected commitment for anchor point %v, requesting "+
+		"delivery", inputCommitment.AnchorPoint)
+
+	resp, err := r.cfg.ChainPorter.RequestSignedDelivery(
```

Comment:
So then this also assumes that the wallet as is will fully handle all the PSBT components? Could imagine another world where there's another external round to sign the Bitcoin portion (as that has a multi-sig/musig2 layer or w/e).

Reply:
Yes, definitely. The BTC level PSBT components will come in a later PR and give you full control over how the anchor TX is created, signed and published.